rust/operator-binary/src/config.rs

```diff
@@ -67,12 +67,12 @@ fn append_ldap_config(config: &mut BTreeMap<String, String>, ldap: &LdapAuthenti
         AirflowConfigOptions::AuthLdapServer.to_string(),
         format!(
             "{protocol}{server_hostname}:{server_port}",
-            protocol = match ldap.tls {
+            protocol = match ldap.tls.tls {
                 None => "ldap://",
                 Some(_) => "ldaps://",
             },
             server_hostname = ldap.hostname,
-            server_port = ldap.port.unwrap_or_else(|| ldap.default_port()),
+            server_port = ldap.port(),
         ),
```

Comment:
@sbernauer I have made the endpoint_url a method, as per https://github.com/stackabletech/operator-rs/pull/705/files, but it is fallable, and this function is not (no return type).

What do you suggest for this PR?
- leave as-is?
- make the return type of this function a `Result<()>`, and fix the callers?

Reply:
For context, see the discussion in the old PR: https://github.com/stackabletech/airflow-operator/pull/364#discussion_r1436814518

Reply:
Haven't looked at the code, but I would prefer to use the new fn and propagate the error correctly